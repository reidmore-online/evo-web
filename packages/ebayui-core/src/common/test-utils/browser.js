import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el, to, cb) {
    fireEvent.scroll(el);
    el.scrollLeft = to;
    setTimeout(cb, 600);
}

function waitFrames(count, cb) {
    if (count) {
        return requestAnimationFrame(() => {
            waitFrames(count - 1, cb);
        });
    }

    cb();
}

async function pressKey(el, info) {
    for (const event of [
        createEvent.keyDown(el, info),
        createEvent.keyUp(el, info),
    ]) {
        // we assign properties to them for older browsers (chrome 49)
        Object.keys(info).forEach((key) => {
            if (event[key] !== info[key]) {
                Object.defineProperty(event, key, { value: info[key] });
            }
        });

        await fireEvent(el, event);
    }
}

const fastAnimations = {
    // Adds an style to the document which forces all transitions to run more quickly for the tests.
    start(disable) {
        if (this.fastAnimationStyle) {
            return;
        }
        const time = disable ? 0 : 1;

        this.fastAnimationStyle = document.createElement("style");
        this.fastAnimationStyle.innerHTML = `*,details::details-content,*::before,*::after {
                animation-duration: ${time}ms !important;
                transition-duration: ${time}ms !important;
                transition-delay: ${time}ms !important;
                scroll-behavior: auto !important;
            }`;
        document.head.appendChild(this.fastAnimationStyle);
    },
    stop() {
        if (!this.fastAnimationStyle) {
            return;
        }

        document.head.removeChild(this.fastAnimationStyle);
        this.fastAnimationStyle = undefined;
    },
};

export { fastAnimations, waitFrames, pressKey, simulateScroll };
