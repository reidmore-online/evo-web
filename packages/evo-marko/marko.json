{
  "taglib-id": "evo-marko",
  "tags-dir": "./src/tags",
  "taglib-imports": [
    "./src/tags/evo-icon/marko.json",
    "./src/tags/evo-skeleton/marko.json"
  ]
}
