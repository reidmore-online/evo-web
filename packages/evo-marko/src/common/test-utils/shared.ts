export function getNItems(n: number, getAttrs: (i: number) => any[]) {
  return Array.from({ length: n }).map((_, i) => getAttrs(i));
}

// Helper method to render pagination items
export function getPaginationItems(
  n: number,
  href: boolean = false,
  selected: number = -1,
  navDisabled: boolean = false,
  skipPrevNext: boolean = false,
) {
  const items = [];
  if (!skipPrevNext) {
    items.push({
      type: "previous",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  for (let i = 0; i < n; i++) {
    items.push({
      href: href ? "#" : null,
      current: i === selected,
    });
  }
  if (!skipPrevNext) {
    items.push({
      type: "next",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  return items;
}
