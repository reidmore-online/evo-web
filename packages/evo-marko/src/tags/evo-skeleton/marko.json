{
  "taglib-id": "evo-marko-skeleton",
  "tags-dir": "./tags"
}
