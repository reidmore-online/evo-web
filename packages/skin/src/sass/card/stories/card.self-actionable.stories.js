export default { title: "Skin/Card/Self Actionable" };

export const verticalButton = () => `
<button class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const verticalLink = () => `
<a class="card" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;

export const horizontalButton = () => `
<button class="card card--horizontal">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const horizontalLink = () => `
<a class="card card--horizontal" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;
