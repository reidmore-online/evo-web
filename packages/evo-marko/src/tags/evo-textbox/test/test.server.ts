import { describe, it } from "vitest";
import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../textbox.stories";

const {
  Isolated,
  WithLabel,
  FloatingLabel,
  Disabled,
  PrefixIcon,
  BothIcons,
  PostfixIcon,
} = composeStories(stories);

describe("evo-textbox", () => {
  it("renders default input textbox", async () => {
    await snapshotHTML(Isolated);
  });

  it("renders default input textbox with an id", async () => {
    await snapshotHTML(Isolated, { id: "textbox-id" });
  });

  it("renders fluid input textbox", async () => {
    await snapshotHTML(Isolated, { fluid: true });
  });

  it("renders a disabled input textbox", async () => {
    await snapshotHTML(Disabled);
  });

  it("renders a input textbox with invalid/error state", async () => {
    await snapshotHTML(Isolated, { invalid: true });
  });

  it("renders a input textbox with external label", async () => {
    await snapshotHTML(WithLabel);
  });

  it("renders a textarea element", async () => {
    await snapshotHTML(Isolated, { multiline: true });
  });

  it("renders a textarea element with prefix icon", async () => {
    await snapshotHTML(PrefixIcon);
  });

  it("renders a textbox element with floating label and without prefix icon", async () => {
    await snapshotHTML(PrefixIcon, { floatingLabel: "test label" });
  });

  it("renders a textarea element with postfix icon", async () => {
    await snapshotHTML(PostfixIcon);
  });

  it("renders a textarea element with postfix icon button", async () => {
    await snapshotHTML(BothIcons);
  });

  it("renders an input textbox with inline floating label", async () => {
    await snapshotHTML(FloatingLabel);
  });

  it("renders an input textbox with opaque floating label", async () => {
    await snapshotHTML(FloatingLabel, { opaqueLabel: true });
  });

  it("renders an input textbox with inline floating label and an id", async () => {
    await snapshotHTML(FloatingLabel, { id: "text-id" });
  });

  it("renders a disabled input textbox with disabled floating label", async () => {
    await snapshotHTML(FloatingLabel, { disabled: true });
  });
});
