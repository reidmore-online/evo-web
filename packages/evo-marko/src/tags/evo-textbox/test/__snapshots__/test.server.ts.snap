// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`evo-textbox > renders a disabled input textbox 1`] = `
<DocumentFragment>
  <span
    class="field"
  >
    <label
      class="field__label field__label--start field__label--disabled"
      for="textbox"
    >
      Email address
    </label>
    <span
      class="textbox textbox--disabled"
      placeholder=""
    >
      <input
        aria-describedby="sM_2 sM_3"
        class="textbox__control"
        disabled=""
        id="textbox"
        type="text"
        value="test"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders a disabled input textbox with disabled floating label 1`] = `
<DocumentFragment>
  <span
    class="floating-label"
  >
    <label
      class="floating-label__label floating-label__label--animate floating-label__label--disabled floating-label__label--inline floating-label__label--disabled"
      for="GENERATED-0"
    >
      Email address
    </label>
    <span
      class="textbox textbox--disabled"
      placeholder=""
    >
      <input
        aria-describedby="sM_2 sM_3"
        class="textbox__control"
        disabled=""
        id="GENERATED-0"
        type="text"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders a input textbox with external label 1`] = `
<DocumentFragment>
  <span
    class="field"
  >
    <label
      class="field__label field__label--start"
      for="textbox"
    >
      Email address
    </label>
    <span
      class="textbox"
      placeholder=""
    >
      <input
        aria-describedby="sM_2 sM_3"
        class="textbox__control"
        id="textbox"
        type="text"
        value="test"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders a input textbox with invalid/error state 1`] = `
<DocumentFragment>
  <span
    class="textbox textbox--invalid"
    placeholder=""
  >
    <input
      aria-describedby="sM_2 sM_3"
      aria-invalid="true"
      class="textbox__control"
      id="GENERATED-0"
      type="text"
    />
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders a textarea element 1`] = `
<DocumentFragment>
  <span
    class="textbox"
    placeholder=""
  >
    <textarea
      aria-describedby="sM_2 sM_3"
      class="textbox__control"
      id="GENERATED-0"
    />
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders a textarea element with postfix icon 1`] = `
<DocumentFragment>
  <span
    class="textbox"
    placeholder=""
  >
    <input
      aria-describedby="sM_2 sM_3"
      class="textbox__control"
      id="GENERATED-0"
      type="text"
    />
    <svg
      aria-hidden="true"
      class="icon icon--24"
    >
      <symbol
        id="GENERATED-1"
        viewBox="0 0 24 24"
      >
        <path
          clip-rule="evenodd"
          d="M16 10a4 4 0 1 1-8 0 4 4 0 0 1 8 0Zm-2 0a2 2 0 1 1-4 0 2 2 0 0 1 4 0Z"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M23 12c0 6.075-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1s11 4.925 11 11Zm-6.404 7.74A8.958 8.958 0 0 1 12 21c-1.68 0-3.25-.46-4.596-1.26C8.147 18.177 9.853 17 12 17c2.147 0 3.853 1.177 4.596 2.74Zm1.62-1.231C17.05 16.424 14.704 15 12 15s-5.05 1.424-6.216 3.508a9 9 0 1 1 12.432 0Z"
          fill-rule="evenodd"
        />
      </symbol>
      <use
        href="#icon-profile-24"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders a textarea element with postfix icon button 1`] = `
<DocumentFragment>
  <span
    class="textbox"
    placeholder=""
  >
    <svg
      aria-hidden="true"
      class="icon icon--24"
    >
      <symbol
        id="GENERATED-0"
        viewBox="0 0 24 24"
      >
        <path
          clip-rule="evenodd"
          d="M16 10a4 4 0 1 1-8 0 4 4 0 0 1 8 0Zm-2 0a2 2 0 1 1-4 0 2 2 0 0 1 4 0Z"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M23 12c0 6.075-4.925 11-11 11S1 18.075 1 12 5.925 1 12 1s11 4.925 11 11Zm-6.404 7.74A8.958 8.958 0 0 1 12 21c-1.68 0-3.25-.46-4.596-1.26C8.147 18.177 9.853 17 12 17c2.147 0 3.853 1.177 4.596 2.74Zm1.62-1.231C17.05 16.424 14.704 15 12 15s-5.05 1.424-6.216 3.508a9 9 0 1 1 12.432 0Z"
          fill-rule="evenodd"
        />
      </symbol>
      <use
        href="#icon-profile-24"
      />
    </svg>
    <input
      aria-describedby="sM_2 sM_3"
      class="textbox__control"
      id="GENERATED-1"
      type="text"
    />
    <svg
      aria-hidden="true"
      class="icon icon--24"
    >
      <symbol
        id="GENERATED-2"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 12C0 5.373 5.373 0 12 0c3.183 0 6.235 1.264 8.485 3.515A11.996 11.996 0 0 1 24 12c0 6.627-5.373 12-12 12S0 18.627 0 12Zm17.968 5.233a.996.996 0 0 0-.32-.989l-4.277-4.27 4.267-4.272a.995.995 0 0 0-1.402-1.404l-4.268 4.271-4.267-4.28a.998.998 0 1 0-1.413 1.413l4.278 4.271-4.268 4.281a.995.995 0 0 0 1.403 1.404l4.267-4.281 4.277 4.27a.994.994 0 0 0 1.723-.414Z"
        />
      </symbol>
      <use
        href="#icon-clear-24"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders a textarea element with prefix icon 1`] = `
<DocumentFragment>
  <span
    class="textbox"
    placeholder=""
  >
    <svg
      aria-hidden="true"
      class="icon icon--24"
    >
      <symbol
        id="GENERATED-0"
        viewBox="0 0 24 24"
      >
        <path
          clip-rule="evenodd"
          d="M4 3a3 3 0 0 0-3 3v12a3 3 0 0 0 3 3h16a3 3 0 0 0 3-3V6a3 3 0 0 0-3-3H4Zm17 3.5V6a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1v.5l8.4 6.3a1 1 0 0 0 1.2 0L21 6.5ZM3 9v9a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1V9l-7.2 5.4a3 3 0 0 1-3.6 0L3 9Z"
          fill-rule="evenodd"
        />
      </symbol>
      <use
        href="#icon-mail-24"
      />
    </svg>
    <input
      aria-describedby="sM_2 sM_3"
      class="textbox__control"
      id="GENERATED-1"
      type="text"
    />
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders a textbox element with floating label and without prefix icon 1`] = `
<DocumentFragment>
  <span
    class="floating-label"
  >
    <label
      class="floating-label__label floating-label__label--animate floating-label__label--inline"
      for="GENERATED-0"
    >
      test label
    </label>
    <span
      class="textbox"
      placeholder=""
    >
      <input
        aria-describedby="sM_2 sM_3"
        class="textbox__control"
        id="GENERATED-0"
        type="text"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders an input textbox with inline floating label 1`] = `
<DocumentFragment>
  <span
    class="floating-label"
  >
    <label
      class="floating-label__label floating-label__label--animate floating-label__label--inline"
      for="GENERATED-0"
    >
      Email address
    </label>
    <span
      class="textbox"
      placeholder=""
    >
      <input
        aria-describedby="sM_2 sM_3"
        class="textbox__control"
        id="GENERATED-0"
        type="text"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders an input textbox with inline floating label and an id 1`] = `
<DocumentFragment>
  <span
    class="floating-label"
  >
    <label
      class="floating-label__label floating-label__label--animate floating-label__label--inline"
      for="text-id"
    >
      Email address
    </label>
    <span
      class="textbox"
      placeholder=""
    >
      <input
        aria-describedby="sM_2 sM_3"
        class="textbox__control"
        id="text-id"
        type="text"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders an input textbox with opaque floating label 1`] = `
<DocumentFragment>
  <span
    class="floating-label floating-label--opaque"
  >
    <label
      class="floating-label__label floating-label__label--animate floating-label__label--inline"
      for="GENERATED-0"
    >
      Email address
    </label>
    <span
      class="textbox"
      placeholder=""
    >
      <input
        aria-describedby="sM_2 sM_3"
        class="textbox__control"
        id="GENERATED-0"
        type="text"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders default input textbox 1`] = `
<DocumentFragment>
  <span
    class="textbox"
    placeholder=""
  >
    <input
      aria-describedby="sM_2 sM_3"
      class="textbox__control"
      id="GENERATED-0"
      type="text"
    />
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders default input textbox with an id 1`] = `
<DocumentFragment>
  <span
    class="textbox"
    placeholder=""
  >
    <input
      aria-describedby="sM_2 sM_3"
      class="textbox__control"
      id="textbox-id"
      type="text"
    />
  </span>
</DocumentFragment>
`;

exports[`evo-textbox > renders fluid input textbox 1`] = `
<DocumentFragment>
  <div
    class="textbox textbox--fluid"
    placeholder=""
  >
    <input
      aria-describedby="sM_2 sM_3"
      class="textbox__control"
      id="GENERATED-0"
      type="text"
    />
  </div>
</DocumentFragment>
`;
